apps/gp2-server/src/data-providers/contentful/utils/members.ts

```diff
@@ -1,41 +1,130 @@
+import {
+  addLocaleToFields,
+  Environment,
+  getLinkEntities,
+  getLinkEntity,
+  patchAndPublish,
+} from '@asap-hub/contentful';
 import { gp2 as gp2Model } from '@asap-hub/model';
 import { GraphQLProject } from '../project.data-provider';
 import { GraphQLWorkingGroup } from '../working-group.data-provider';
+import { getIdsToDelete } from './common';
 
 type MembersItem =
   | GraphQLWorkingGroup['membersCollection']
   | GraphQLProject['membersCollection'];
 
 type MemberItem = NonNullable<NonNullable<MembersItem>['items'][number]>;
 const parseMember = <T extends string>(
+  id: string,
   user: NonNullable<MemberItem['user']>,
   role: MemberItem['role'],
 ): {
+  id: string;
   userId: string;
   role: T;
   firstName: string;
   lastName: string;
   avatarUrl?: string;
 } => ({
+  id,
   userId: user.sys.id,
   role: (role ?? '') as T,
   firstName: user.firstName ?? '',
   lastName: user.lastName ?? '',
   avatarUrl: user.avatar?.url ?? undefined,
 });
 
-export const parseMembers = <T extends string>(
-  members:
-    | GraphQLWorkingGroup['membersCollection']
-    | GraphQLProject['membersCollection'],
-) =>
+export const parseMembers = <T extends string>(members: MembersItem) =>
   members?.items
     .filter((member): member is MemberItem => member !== null)
     .reduce((membersList: gp2Model.Member<T>[], member) => {
       const user = member?.user;
       if (!user?.onboarded) {
         return membersList;
       }
-      const groupMember = parseMember<T>(user, member.role);
+      const groupMember = parseMember<T>(member.sys.id, user, member.role);
       return [...membersList, groupMember];
     }, []) || [];
+
+const addNextMember = async <T extends string>(
+  environment: Environment,
```

Comment:
looks to me like all these could be members of a class which takes environment as a dependency, or they could be part of the data provider which has that environment in the scope